pymeasure/instruments/siglent/sdl1022x.py

```diff
@@ -0,0 +1,213 @@
+        load.input_state = "ON"
+        print(load.voltage)
+
+    """
+    voltage = Instrument.measurement(
+        "MEASure:VOLT:DC?",
+        "Voltage across the load in Volts"
+    )
+
+    current = Instrument.measurement(
+        "MEASure:CURR:DC?",
+        "Current across the load in Amps"
+    )
+
+    power = Instrument.measurement(
+        "MEASure:POW:DC?",
+        "Power across the load in Watts"
+    )
+
+    resistance = Instrument.measurement(
+        "MEASure:RES:DC?",
+        "Resistance across the load in Ohms"
+    )
+
+    external = Instrument.measurement(
+        "MEASure:EXT?",
+        "Voltage across the external inputs in Volts"
+    )
+
+    input_state = Instrument.control(
+        ":SOURce:INPut:STATe?",
+        ":SOURce:INPut:STATe %s",
+        "Set or query the load state",
+        map_values=True,
+        validator=strict_discrete_set,
+        values={True: 1, 'on': 1, 'ON': 1, False: 0, 'off': 0, 'OFF': 0}
+    )
+
+    short_state = Instrument.control(
+        ":SOURce:SHORt:STATe?",
+        ":SOURce:SHORt:STATe %s",
+        "Set or query the short circuit status",
+        map_values=True,
+        validator=strict_discrete_set,
+        values={True: 1, 'on': 1, 'ON': 1, False: 0, 'off': 0, 'OFF': 0}
+    )
+
+    transient_mode = Instrument.control(
+        ":SOURce:FUNCtion:TRANsient?",
+        "SOURce:FUNCtion:TRANsient %s",
+        "Set or query the transient load mode",
+        map_values=True,
+        validator=strict_discrete_set,
+        values=VALID_LOAD_MODES_DICT
+    )
+
+    load_mode = Instrument.control(
+        ":SOURce:FUNC?",
+        ":SOURce:FUNC %s",
+        "Which mode the electronic load is in",
+        validator=strict_discrete_set,
+        values=["CURRENT", "VOLTAGE", "POWER", "RESISTANCE", "LED"]
+    )
+
+    constant_current_load = Instrument.control(
+        ":SOUR:CURR?",
+        ":SOUR:CURR %g",
+        "Load current in amps",
+        validator=strict_range,
+        values=[0, 30]
+    )
+
+    constant_voltage_load = Instrument.control(
+        ":SOUR:VOLT?",
+        ":SOUR:VOLT %g",
+        "Load voltage in volts",
+        validator=strict_range,
+        values=[0, 150]
+    )
+
+    constant_resistance_load = Instrument.control(
+        ":SOUR:RES?",
+        ":SOUR:RES %g",
+        "Load resistance in Ohms",
+        validator=strict_range,
+        values=[0, 10000]
+    )
+    constant_power_load = Instrument.control(
+        ":SOUR:POW?",
+        ":SOUR:POW %g",
+        "Load power in Watts",
+        validator=strict_range,
+        values=[0, 200]
+    )
+
+    def __init__(self, adapter, **kwargs):
+        super(Sdl1022X, self).__init__(
+            adapter, "Siglent SDL1022X Electronic Load", **kwargs
+        )
+
+    def set_current_range(self, mode, current):
+        if current > max(VALID_CURRENT_RANGES):
+            raise ValueError
+        if mode not in VALID_LOAD_MODES_DICT.keys():
```

Comment:
By the way, you can reuse the same [validators](https://pymeasure.readthedocs.io/en/latest/api/instruments/validators.html) as for the property definitions (e.g. ` strict_discrete_set`) to do the value testing for you.